pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -461,7 +461,22 @@ func buildInboundCatchAllNetworkFilterChains(configgen *ConfigGeneratorImpl,
 	if node.SupportsIPv6() {
 		ipVersions = append(ipVersions, util.InboundPassthroughClusterIpv6)
 	}
-	filterChains := make([]*listener.FilterChain, 0, 2)
+	filterChains := make([]*listener.FilterChain, 0, 3)
+	if features.PilotEnableLoopBlockers {
+		filterChains = append(filterChains, &listener.FilterChain{
+			Name: VirtualInboundBlackholeFilterChainName,
+			FilterChainMatch: &listener.FilterChainMatch{
```

Comment:
Can we create this once and reuse?

Reply:
I was going to but I am concerned if an EnvoyFilter modifies it and it's shared then every push we will apply the same modification. In similar code in 1.4 we saw this where an access log filter got appended every push and ended up with millions of access log filters...

I am also concerned that's are other places in our code with this same problem

Reply:
Sure